secrets/awskms/kms_test.go

```diff
@@ -57,3 +55,27 @@ func newHarness(ctx context.Context, t *testing.T) (drivertest.Harness, error) {
 func TestConformance(t *testing.T) {
 	drivertest.RunConformanceTests(t, newHarness)
 }
+
+// KMS-specific tests.
+
+func TestNoConnectionError(t *testing.T) {
+	if _, err := Dial(nil); err == nil {
+		t.Error("got nil, want no AWS session provided")
+	}
+
+	os.Setenv("AWS_ACCESS_KEY", "myaccesskey")
+	os.Setenv("AWS_SECRET_KEY", "mysecretkey")
+	os.Setenv("AWS_REGION", "us-east-1")
+	sess, err := session.NewSession()
+	if err != nil {
+		t.Fatal(err)
+	}
+	ctx := context.Background()
```

Comment:
Nit: `ctx` and `plaintext` are only used once, so just use them inline, it's easier to read.

Reply:
Done.